Dependency verification breaks init scripts with external dependencies

<!--- 
Please follow the instructions below. 
We receive dozens of issues every week, so to stay productive, we will close issues that don't provide enough information. 

Please open Android-related issues on the Android Issue Tracker at https://source.android.com/source/report-bugs
Please open Gradle Native-related issues at https://github.com/gradle/gradle-native/issues
-->

<!--- Provide a brief summary of the issue in the title above -->
Running the example from
[init scripts with external dependencies](https://docs.gradle.org/current/userguide/init_scripts.html#sec:custom_classpath):
`./gradlew --init-script init.gradle -q doNothing`
is not possible if dependency verification is turned on.

### Expected Behavior
<!--- Tell us what should happen -->
Init scripts to still work when dependency verification is turned on, or at least fail with more information.

### Current Behavior
<!--- Tell us what happens instead of the expected behavior -->
```
./gradlew --init-script init.gradle -q doNothing

FAILURE: Build failed with an exception.

* What went wrong:
GradleProperties has not been loaded yet.
```

### Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
I have a few common plugins that I enable in all my gradle projects by using $HOME/.gradle/init.gradle, I either have to disable dependency verification or remove the init script when I work on a project that uses dependency verification.

### Steps to Reproduce 
<!--- Provide a self-contained example project (as an attached archive or a Github project). -->
<!--- In the rare cases where this is infeasible, we will also accept a detailed set of instructions. -->
Example project: https://github.com/thejk/gradle-init-script-issue
Just run: `./gradlew --init-script init.gradle -q doNothing`

### Your Environment
openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment (build 11.0.8+10)
OpenJDK 64-Bit Server VM (build 11.0.8+10, mixed mode)

--- a/subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationIntegrityCheckIntegTest.groovy
+++ b/subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationIntegrityCheckIntegTest.groovy
@@ -1055,4 +1055,40 @@ This can indicate that a dependency has been compromised. Please carefully verif
 """
     }
 
+    def "reasonable error message for dependencies of init scripts which are missing from verification file"() {
+        file("init.gradle") << """
+            initscript {
+                repositories {
+                    maven { url "${mavenHttpRepo.uri}" }
+                }
+                dependencies {
+                    classpath 'org:foo:1.0'
+                }
+            }
+        """
+
+        def mod = mavenHttpRepo.module('org', 'foo', '1.0')
+            .publish()
+
+        createMetadataFile {
+        }
+
+        buildFile << """
+            tasks.register("noop") {
+            }
+        """
+
+        when:
+        mod.pom.expectGet()
+        mod.artifact.expectGet()
+        executer.withArguments("-I", "init.gradle")
+        fails 'noop'
+
+        then:
+        outputContains "Gradle properties are not loaded yet, any customization to dependency verification will be ignored until the main build script is loaded."
+        failure.assertHasDescription """Dependency verification failed for configuration 'classpath'
+2 artifacts failed verification:
+  - foo-1.0.jar (org:foo:1.0) from repository maven
+  - foo-1.0.pom (org:foo:1.0) from repository maven"""
+    }
 }

--- a/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/verification/report/DependencyVerificationReportWriter.java
+++ b/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/verification/report/DependencyVerificationReportWriter.java
@@ -23,6 +23,8 @@ import org.gradle.api.internal.artifacts.verification.verifier.MissingChecksums;
 import org.gradle.api.internal.artifacts.verification.verifier.SignatureVerificationFailure;
 import org.gradle.api.internal.artifacts.verification.verifier.VerificationFailure;
 import org.gradle.api.internal.properties.GradleProperties;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
 import org.gradle.internal.Factory;
 import org.gradle.internal.component.external.model.ModuleComponentArtifactIdentifier;
 
@@ -34,6 +36,8 @@ import java.util.List;
 import java.util.Map;
 
 public class DependencyVerificationReportWriter {
+    private static final Logger LOGGER = Logging.getLogger(DependencyVerificationReportWriter.class);
+
     private static final Comparator<Map.Entry<ModuleComponentArtifactIdentifier, Collection<RepositoryAwareVerificationFailure>>> DELETED_LAST = Comparator.comparing(e -> e.getValue().stream().anyMatch(f -> f.getFailure() instanceof DeletedArtifact) ? 1 : 0);
     private static final Comparator<Map.Entry<ModuleComponentArtifactIdentifier, Collection<RepositoryAwareVerificationFailure>>> MISSING_LAST = Comparator.comparing(e -> e.getValue().stream().anyMatch(f -> f.getFailure() instanceof MissingChecksums) ? 1 : 0);
     private static final Comparator<Map.Entry<ModuleComponentArtifactIdentifier, Collection<RepositoryAwareVerificationFailure>>> BY_MODULE_ID = Comparator.comparing(e -> e.getKey().getDisplayName());
@@ -59,8 +63,15 @@ public class DependencyVerificationReportWriter {
     }
 
     private static boolean isVerboseConsoleReport(GradleProperties gradleProperties) {
-        String param = gradleProperties.find(VERBOSE_CONSOLE);
-        return VERBOSE_VALUE.equals(param);
+        try {
+            String param = gradleProperties.find(VERBOSE_CONSOLE);
+            return VERBOSE_VALUE.equals(param);
+        } catch (IllegalStateException e) {
+            // Gradle properties are not loaded yet, which can happen in init scripts
+            // let's return a default value
+            LOGGER.warn("Gradle properties are not loaded yet, any customization to dependency verification will be ignored until the main build script is loaded.");
+            return false;
+        }
     }
 
     private AbstractTextDependencyVerificationReportRenderer createConsoleRenderer(Path gradleUserHome, DocumentationRegistry documentationRegistry, GradleProperties gradleProperties) {

